src/Cech_complex/utilities/delaunay_cech_persistence.cpp

```diff
@@ -0,0 +1,159 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2024 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <gudhi/Alpha_complex.h>
+#include <gudhi/MEB_filtration.h>
+#include <gudhi/Simplex_tree.h>
+#include <gudhi/Persistent_cohomology.h>
+#include <gudhi/Points_off_io.h>
+
+#include <boost/program_options.hpp>
+
+#include <CGAL/Epeck_d.h>  // For EXACT or SAFE version
+#include <CGAL/Epick_d.h>  // For FAST version
+
+#include <string>
+#include <vector>
+#include <limits>  // infinity
+
+// Types definition
+using Simplex_tree = Gudhi::Simplex_tree<>;
+using Filtration_value = Simplex_tree::Filtration_value;
+
+using Field_Zp = Gudhi::persistent_cohomology::Field_Zp;
+using Persistent_cohomology = Gudhi::persistent_cohomology::Persistent_cohomology<Simplex_tree, Field_Zp>;
+
+void program_options(int argc, char* argv[], std::string& off_file_points, bool& exact, bool& fast,
+                     std::string& pers_diag_file, Filtration_value& max_radius, int& p,
+                     Filtration_value& min_persistence);
+
+template<class Kernel>
+Simplex_tree create_simplex_tree(const std::string &off_file_points, bool exact_version, Filtration_value max_radius) {
+  using Point = typename Kernel::Point_d;
+  using Points_off_reader = Gudhi::Points_off_reader<Point>;
+  using Delaunay_complex = Gudhi::alpha_complex::Alpha_complex<Kernel>;
+
+  Simplex_tree stree;
+
+  Points_off_reader off_reader(off_file_points);
+  std::vector<Point> point_cloud = off_reader.get_point_cloud();
+  Delaunay_complex delaunay_complex_from_file(point_cloud);
+  delaunay_complex_from_file.create_complex(stree, std::numeric_limits< Filtration_value >::infinity(),
+                                            // exact can be false (or true), as default_filtration_value is set to true
+                                            false, true);
+  Gudhi::cech_complex::assign_MEB_filtration(Kernel(), stree, point_cloud, exact_version);
+  stree.prune_above_filtration(max_radius);
```

Comment:
Unlike the alpha-complex, for Delaunay-Cech, we could do a bit better (don't compute filtration for simplices whose faces already have a too large filtration), but that's not important for now.